velox/exec/RowNumber.cpp

```diff
@@ -75,7 +75,9 @@ void RowNumber::addInput(RowVectorPtr input) {
   const auto numInput = input->size();
 
   if (table_) {
-    ensureInputFits(input);
+    if (inputSpiller_ == nullptr) {
```

Comment:
nit: let's check this inside ensureInputFits? Thanks!

Reply:
ensureInputFits() is called in multiple locations and the condition for skipping is different.